solutions/src/main/java/ru/sberbank/school/task06/CollectionUtils.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /**
+     * Обрезание списка заданным числом элементов. limit({1, 2 ,3}, 2) => {1, 2}
+     *
+     * @param source список
+     * @param size   необходимое число элементов
+     * @return сокращенный список
+     */
+    public static <T> List<T> limit(List<T> source, int size) {
+        if(source.size() == 0 || source.size() < size) {
+            return source;
+        }
+        return source.subList(0, size);
+    }
+
+    /**
+     * Добавление элемента в коллекцию.
+     *
+     * @param dest коллекция, в которую необходимо вставить элемент
+     * @param o    элемент
+     */
+    public static <T> void add(List<? super T> dest, T o) {
+        dest.add(o);
+    }
+
+
+    /**
+     * Удаление всеъх элементов одной коллекии из другой.
+     *
+     * @param removeFrom коллекция, из которой удалять элементы
+     * @param toRemove   коллекция, в которой лежат элементы для удаления из коллекции removeFrom
+     */
+    public static void removeAll(List<?> removeFrom, List<?> toRemove) {                               //???
+        Iterator iterator = toRemove.iterator();
+        while (iterator.hasNext()) {
+            Object element = iterator.next();
+            if (removeFrom.contains(element)) {
+                iterator.remove();
+            }
+        }
+
+    }
+
+    /**
+     * Проверка на содержение одной коллекции в другой.
+     *
+     * @param c1 первая коллекция
+     * @param c2 вторая коллекция
+     * @return true, если все элементы коллекции c2 содержатся в c1
+     */
+    public static <T> boolean containsAll(List<?> c1, List<?> c2) {
+        return c1.containsAll(c2);
+    }
+
+    /**
+     * Проверка на содержение хотя бы одного элемента одной коллекции в другой.
+     *
+     * @param c1 первая коллекция
+     * @param c2 вторая коллекция
+     * @return true, если хотя бы 1 элемент коллекции c2 содержатся в c1
+     */
+    public static <T> boolean containsAny(List<?> c1, List<?> c2) {
+        for (Object element : c2) {
+            if (c1.contains(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Создание коллекции, содержащей элементы из входной коллекции в диапазоне от min до max (не
+     * индексы, а значения). Элементы необходимо сравнивать, через интерфейс Comparable Пример:
+     * range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6) вернет {3,4,5,6}
+     *
+     * @param list исходная коллекция
+     * @param min  минимальное значение
+     * @param max  максимальнео значение
+     * @return отфильрованный по минимальному и максимальному значению список
+     */
+    public static <T extends Comparable<? super T>> List<T> range(List<? extends T> list, T min, T max) {
+         List<T> newlist = new ArrayList<>();
+         for (T element : list) {
+             if (element.compareTo(min) >= 0 && element.compareTo(max) <= 0) {
+                 if (!newlist.isEmpty()) {
+                     int index = newlist.size();
```

Comment:
&#39;if&#39; child has incorrect indentation level 21, expected level should be 20.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1790646113>*